agents/agents/executor/export_for_testing.go

```diff
@@ -32,20 +30,10 @@ import (
 //nolint:cyclop
 func NewExecutorInjectedBackend(ctx context.Context, config config.Config, executorDB db.ExecutorDB, scribeClient client.ScribeClient, clients map[uint32]Backend, urls map[uint32]string, handler metrics.Handler) (*Executor, error) {
 	chainExecutors := make(map[uint32]*chainExecutor)
-	conn, err := grpc.DialContext(ctx, fmt.Sprintf("%s:%d", scribeClient.URL, scribeClient.Port), grpc.WithTransportCredentials(insecure.NewCredentials()))
-	if err != nil {
-		return nil, fmt.Errorf("could not dial grpc: %w", err)
-	}
-
-	grpcClient := pbscribe.NewScribeServiceClient(conn)
 
-	// Ensure that gRPC is up and running.
-	healthCheck, err := grpcClient.Check(ctx, &pbscribe.HealthCheckRequest{}, grpc.WaitForReady(true))
+	conn, grpcClient, err := makeScribeClient(ctx, handler, fmt.Sprintf("%s:%d", scribeClient.URL, scribeClient.Port))
```

Comment:
unrelated, but I hate this scribeclient struct pattern. Just kinda wish it would handle more of this otel stuff for you. Not related to this pr